batch/batch/front_end/front_end.py

```diff
@@ -1551,6 +1545,22 @@ async def commit_update(request: web.Request, userdata):
 
 async def _commit_update(app: web.Application, batch_id: int, update_id: int, user: str, db: Database):
     client_session: httpx.ClientSession = app['client_session']
+
+    # We don't allow updates to batches that have been cancelled to be committed
+    record = await db.execute_and_fetchone(
+        '''
+SELECT cancelled
+FROM batches
+WHERE id = %s AND user = %s AND NOT deleted
+FOR UPDATE;
+''',
+        (batch_id, user),
+    )
+    if not record:
+        raise web.HTTPNotFound()
+    if record['cancelled']:
+        raise web.HTTPBadRequest(reason='Cannot commit and update to a cancelled batch')
```

Comment:
Can you instead of adding this query also grab `cancelled` in the query of the `commit_update` function (the one that is actually the REST endpoint)? This function is used in multiple places including places like `create-fast` where this is redundant. I'd rather consolidate authorization checks like these as close to the endpoint as possible so these internal reused functions don't pile on redundant checks to the lifetime of a request.